src/nvim/autocmd.c

```diff
@@ -105,17 +105,28 @@ static char_u *old_termresponse = NULL;
 #define FOR_ALL_AUPATS_IN_EVENT(event, ap) \
   for (AutoPat *ap = first_autopat[event]; ap != NULL; ap = ap->next)  // NOLINT
 
-// Map of autocmd group names.
+// Map of autocmd group names and ids.
 //  name -> ID
-static Map(String, int) augroup_map = MAP_INIT;
+//  ID -> name
+static Map(String, int) map_augroup_name_to_id = MAP_INIT;
+static Map(int, String) map_augroup_id_to_name = MAP_INIT;
 
-static void augroup_map_del(char *name)
+static void augroup_name_map_del(char *name)
 {
-  String key = map_key(String, int)(&augroup_map, cstr_as_string(name));
-  map_del(String, int)(&augroup_map, key);
+  String key = map_key(String, int)(&map_augroup_name_to_id, cstr_as_string(name));
+  map_del(String, int)(&map_augroup_name_to_id, key);
   api_free_string(key);
 }
 
+static void augroup_id_map_del(int id)
```

Comment:
I'd rather have only one function and pass an ID less than 0 to it to skip deleting the ID from the map. I don't like having two functions here, since it's very easy to forget to use both in the future if we have to make more modifications.

Reply:
Actually, I want the entry to be deleted from id->name map, and _not_ from name->id map.

I could pass in a `NULL` for `name` parameter and put a conditional for it.